File: solutions/1572. Matrix Diagonal Sum/1572.py
# Time complexity: O(n)
# Space complexity: O(1)
class Solution:
  def diagonalSum(self, mat: List[List[int]]) -> int:
    n = len(mat)
    ans = 0

    for i in range(n):
      ans += mat[i][i]
      ans += mat[n - 1 - i][i]

    return ans if n % 2 == 0 else ans - mat[n // 2][n // 2]

File: solutions/1572. Matrix Diagonal Sum/1572.rs
// Time complexity: O(n)
// Space complexity: O(1)
impl Solution {
    pub fn diagonal_sum(mat: Vec<Vec<i32>>) -> i32 {
        let n = mat.len();
        let mut ans = 0;

        for i in 0..n {
            ans += mat[i][i];
            ans += mat[n - 1 - i][i];
        }

        if n % 2 == 0 {
            ans
        } else {
            ans - mat[n / 2][n / 2]
        }
    }
}
